nyx_client/nyx_client/data.py

```diff
@@ -54,6 +54,8 @@ class Data:
     custom_metadata: list[Property]
     """Additional metadata properties to decorate the data with. Note that nyx-internal properties are not allowed.
     """
+    connection_id: str
+    """The ID of the connection"""
```

Comment:
(pedantic - also add link, like in client, since it's a string here unlike e.g. with `custom_metadata` above which auto-links to Property due to typing)